src/Command/VarnishControllerDeployCommand.php

```diff
@@ -0,0 +1,160 @@
+            ->addOption('username', '', InputOption::VALUE_REQUIRED, 'VAC username [required]')
+            ->addOption('password', '', InputOption::VALUE_REQUIRED, 'VAC password [required]')
+            ->addOption('vcl-name', '', InputOption::VALUE_REQUIRED, 'VCL name [required]')
+            ->addOption('vcl-group', '', InputOption::VALUE_REQUIRED, 'VCL group [required]')
+            ->addOption('verify-tls', '', InputOption::VALUE_REQUIRED, 'Specifies TLS verification, true|false|/path/to/certificate. See http://docs.guzzlephp.org/en/stable/request-options.html#verify for possible options', 'true')
+        ;
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->logger = new ConsoleLogger($output, [
+            LogLevel::INFO => OutputInterface::VERBOSITY_NORMAL,
+        ]);
+
+        $vclInput = $this->getArgumentString($input, 'vcl');
+        $uri = $this->requireStringOption($input, 'uri');
+        $username = $this->requireStringOption($input, 'username');
+        $password = $this->requireStringOption($input, 'password');
+        $vclName = $this->requireStringOption($input, 'vcl-name');
+        $vclGroup = $this->requireStringOption($input, 'vcl-group');
+        $verifyTLS = $this->convertToBoolean($this->requireStringOption($input, 'verify-tls'));
+
+        $this->client = new VarnishControllerClient($uri, $username, $password, $verifyTLS);
+        $rollbackID = '';
+
+        $groupID = $this->client->getGroupID($vclGroup);
+
+        $vclFile = $this->initVclFile();
+
+        $newVclSource = $this->readFile($vclInput);
+
+        if (!$newVclSource) {
+            throw new \InvalidArgumentException(sprintf('Filename "%s" is not readable.', $vclInput));
+        }
+
+        if ($newVclSource === $vclFile->getSource()) {
+            $this->logger->log('info', 'VCL did not change, will not update');
+        } else {
+            //todo rollback
+            //$rollbackID = $this->client->getHead($vclFile->getId());
+            //$this->>logger->log('info', 'Using VCL {id} which head is {rollbackID}', ['id' => $vclFile->getId(), 'rollbackID' => $rollbackID]);
+
+            $vclCommit = $this->client->updateVCL($vclFile->getId(), $newVclSource);
+            $this->logger->log('info', 'Pushed new VCL with the commit ID {id}', ['id' => $vclCommit]);
+        }
+
+
+        if (!$groupID) {
+            throw new InvalidOptionException($vclGroup.' needs to be a valid group name');
+        }
+
+        $this->logger->log(
+            'info', 'Deploying VCL {id} to group {groupID}', ['id' => $vclFile->getId(), 'groupID' => $groupID]);
+
+        try {
+            $this->client->deploy($groupID, $vclFile->getId());
+            $this->logger->log('info', 'Successfully deployed VCL {id} to group {groupID}', ['id' => $vclFile->getId(), 'groupID' => $groupID]);
```

Comment:
why not use the `LoggerInterface::info` method anymore? i fnd that easier to read then passing the verbosity as string.